tools/junitreport/pkg/parser/gotest/parser.go

```diff
@@ -0,0 +1,100 @@
+package gotest
+
+import (
+	"bufio"
+	"strings"
+
+	"github.com/openshift/origin/tools/junitreport/pkg/api"
+	"github.com/openshift/origin/tools/junitreport/pkg/builder"
+	"github.com/openshift/origin/tools/junitreport/pkg/parser"
+)
+
+// NewParser returns a new parser that's capable of parsing Go unit test output
+func NewParser(builder builder.TestSuitesBuilder) parser.TestOutputParser {
+	return &testOutputParser{
+		builder:     builder,
+		testParser:  newTestDataParser(),
+		suiteParser: newTestSuiteDataParser(),
+	}
+}
+
+type testOutputParser struct {
+	builder     builder.TestSuitesBuilder
+	testParser  parser.TestDataParser
+	suiteParser parser.TestSuiteDataParser
+}
+
+// Parse parses `go test -v` output into test suites. Test output from `go test -v` is not bookmarked for packages, so
+// the parsing strategy is to advance line-by-line, building up a slice of test cases until a package declaration is found,
+// at which point all tests cases are added to that package and the process can start again.
+func (p *testOutputParser) Parse(input *bufio.Scanner) (*api.TestSuites, error) {
+	currentSuite := &api.TestSuite{}
+	var currentTest *api.TestCase
+	var currentTestResult api.TestResult
+	var currentTestOutput []string
+
+	for input.Scan() {
+		line := input.Text()
+		isTestOutput := true
+
+		if p.testParser.MarksBeginning(line) || p.suiteParser.MarksCompletion(line) {
+			if currentTest != nil {
+				// we can't mark the test as failed or skipped until we have all of the test output, which we don't know
+				// we have until we see the next test or the beginning of suite output, so we add it here
+				output := strings.Join(currentTestOutput, "\n")
+				switch currentTestResult {
+				case api.TestResultSkip:
+					currentTest.MarkSkipped(output)
+				case api.TestResultFail:
+					currentTest.MarkFailed("", output)
+				}
+
+				currentSuite.AddTestCase(currentTest)
+			}
+			currentTest = &api.TestCase{}
+			currentTestResult = api.TestResultFail
+			currentTestOutput = []string{}
```

Comment:
Is it intended that if my passing test spits some output eg.:
```
$ hack/test-go.sh pkg/build/controller/ -v -cover
=== RUN   TestHandleBuildConfig
E0105 11:11:52.973414    4347 config_controller.go:66] error instantiating Build from BuildConfig /testBuildConfig: error
--- PASS: TestHandleBuildConfig (0.00s)
...
```
You're ignore this, and only skipped and failed output is part of the report?